code/game/objects/items/handcuffs.dm

```diff
@@ -546,21 +550,34 @@
 /obj/item/restraints/legcuffs/bola/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	if(..() || !iscarbon(hit_atom))//if it gets caught or the target can't be cuffed,
 		return//abort
-	ensnare(hit_atom)
+	// Update the flag so we will ensnare the mob after the after_throw proc
+	is_mob_hit = TRUE
+	ensnare_mob_ref = WEAKREF(hit_atom)
+
+/obj/item/restraints/legcuffs/bola/after_throw(datum/callback/callback)
+	. = ..()
+	if (!is_mob_hit && !ensnare_mob_ref)
+		return
+	var/atom/ensnare_mob = ensnare_mob_ref.resolve()
+	if (is_mob_hit && !isnull(ensnare_mob))
+		ensnare(ensnare_mob)
```

Comment:
Same here, you can just check if ensnare_mob_ref is not null in the first check and ensure that the atom exists in the second one

Reply:
Yup, did just that